### PR TITLE
fix regression to allow for empty-string paths

### PR DESCRIPTION
**Pull request by heygrady**
fixes https://github.com/ReactTraining/react-router/issues/6941
related https://github.com/ReactTraining/react-router/pull/6715

### CODE REVIEW

File: packages/react-router/modules/matchPath.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ function matchPath(pathname, options = {}) {
   const paths = [].concat(path);
 
   return paths.reduce((matched, path) => {
-    if (!path) return null;
+    // https://github.com/ReactTraining/react-router/issues/6941
```
Comment (timdorr) [inline packages/react-router/modules/matchPath.js]:
```suggestion
```
The link to this will be in their history, so this comment isn't needed.

Comment (heygrady) [reply packages/react-router/modules/matchPath.js]: removed

**Comment by timdorr** [comment]
Thanks!

**Comment by mjackson** [comment]
Thanks for getting this merged, @timdorr. I tried to push a 5.1.1 release earlier today to fix the `<Link ref>` issue, but [the release failed](https://travis-ci.org/ReactTraining/react-router/jobs/590516835). I can't reproduce the failure locally, so I'm going to have to debug this afternoon. I'll push this fix in 5.1.1 and update the website when I'm done.

**Pull request merged**